server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2224,6 +2232,21 @@ else if (details != null && !details.isEmpty()) {
         return _tmpltDao.findById(id);
     }
 
+    private void copyTemplateToAllZones(Long id) {
+        s_logger.info("Copying template to all zones since it will be a system template");
+        List<DataCenterVO> zones = _dcDao.listEnabledZones();
+        if (zones != null) {
+            List<Long> zoneIds = zones.stream().map(zone -> zone.getId()).collect(Collectors.toList());
```

Comment:
Should we skip the zone on which the template is already in?

Reply:
this seems to be handled by https://github.com/apache/cloudstack/blob/c59d0c82137f2294c67d4dd77dd1f4a69b2edb09/server/src/main/java/com/cloud/template/TemplateManagerImpl.java#L931-L936